src/helpers.ts

```diff
@@ -427,3 +429,21 @@ export function tagsMatch(tags, pattern) {
   const matchess = micromatch(tags || ['*'], pattern);
   return matchess.length > 0;
 }
+
+export function filterBrowserConsoles(
+  messages: BrowserMessage[],
+  status: StatusValue
+) {
+  if (status === 'failed') {
+    return messages;
+  }
+  if (status === 'succeeded') {
```

Comment:
couple of optimizations we can do here.
1. if message is less than 100, return without doing unnecessary work.
2. do one pass across the messages and based on types, construct the final messages list with giving priority to error's and warnings.

Reply:
first make sense, but isn't second micro optimisation?
i feel readability is more important compared to micro optimisation

Reply:
I have done the first